torch/_functorch/_aot_autograd/autograd_cache.py

```diff
@@ -280,6 +280,63 @@ def check_cacheable(gm: torch.fx.GraphModule):
         check_cacheable(gm.saved_tensors_hooks_unpack_0)  # type: ignore[arg-type]
 
 
+def _get_context_fn_cache_hash(context_fn):
+    """
+    Extract a cache hash from a context_fn used for selective activation checkpointing (SAC).
+
+    The context_fn determines which ops are saved vs recomputed in the SAC region.
+    Since context_fn can be an arbitrary Python function, we cannot reliably pickle
+    it for cache key generation (pickle only captures the function name, not the code).
+
+    Users can provide a stable hash by setting a `cache_hash` attribute on the context_fn
+    (or on the underlying policy function for functools.partial objects).
+
+    Returns:
+        str: The cache hash if found
+        None: If no hash is provided (caller should bypass caching)
+    """
+    import functools
+
+    if hasattr(context_fn, "cache_hash"):
+        return context_fn.cache_hash
+
+    if isinstance(context_fn, functools.partial):
+        if hasattr(context_fn.func, "cache_hash"):
+            return context_fn.func.cache_hash
+        for arg in context_fn.args:
```

Comment:
You shouldn't be iterating over every argument, as only the first argument has a hash. Actually, I think you should just only test `func` here.

Reply:
Oh, I see why you are doing this. But this is still bad. You need to test for the specific helper builder function, and when you do that test you should incorporate that into the hash (because there may be potentially different builder functions with different semantics)